src/sentry/workflow_engine/models/data_condition_group.py

```diff
@@ -0,0 +1,28 @@
+from django.db import models
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import DefaultFieldsModel, region_silo_model, sane_repr
+
+from .detector import Detector
+from .workflow_action import WorkflowAction
+
+
+@region_silo_model
+class DataConditionGroup(DefaultFieldsModel):
+    """
+    A data condition is a way to specify a condition that must be met for a workflow to execute.
+    """
+
+    __relocation_scope__ = RelocationScope.Organization
+    __repr__ = sane_repr("type", "condition")
+
+    class Type(models.TextChoices):
+        ANY = "any"
+        ALL = "all"
+
+    logic_type = models.CharField(max_length=200, choices=Type.choices, default=Type.ANY)
+
+    detector = models.ForeignKey(Detector, on_delete=models.CASCADE, blank=True, null=True)
+    workflow_action = models.ForeignKey(
+        WorkflowAction, on_delete=models.CASCADE, blank=True, null=True
+    )
```

Comment:
Shouldn't we allow multiple workflow actions to be linked here? So this group of conditions fires this group of actions?